### PR TITLE
[v1.x] use gcc-8.5 instead of gcc-10 for aarch64 build

### PR DESCRIPTION
**Pull request by mseth10**
## Description ##
In order to get the advantage of LSE support while supporting all AArch64 platforms, we used gcc-10 on u18 base docker image as it supports the build flag `-moutline-atomics`. But using gcc-10 also creates a dependency of generated MXNet wheel on GLIBCXX_3.4.26, and so we need to upgrade libstdc++6 package on u18 to get the wheel working. On u20, the wheel works fine.

2 weeks ago gcc-8.5 was launched that added support for the build flag `-moutline-atomics`. Since there is no ubuntu/canonical package for gcc-8.5, I tried building it from source and it works, with the advantage over gcc-10 that we dont need to upgrade libstdc++6 on u18.

### Changes ###
- [x] Add LSE support to glibc with the package `libc6-lse`.
- [x] Use build flag `-march=armv8-a+crc+crypto` instead of `-march=armv8` earlier to get the benefit of crc and crypto extension.
- [x] Install gcc-8 (gcc-8.4) as openblas build fails with gcc-8.5 and so using gcc-8.4 to build it.

### CODE REVIEW

**Review by Zha0q1 — APPROVED** [review]
LGTM!

**Comment by mseth10** [comment]
@mxnet-bot run ci [centos-gpu]

**Comment by mseth10** [comment]
@mxnet-bot run ci [unix-gpu]

File: tools/dependencies/openblas.sh [hunk 1]
```diff
@@ -33,7 +33,12 @@ if [[ ((! -e $DEPS_PATH/lib/libopenblas.a) && -z "$CMAKE_STATICBUILD") ||
     cd $DEPS_PATH/OpenBLAS-$OPENBLAS_VERSION
 
     # Adding NO_DYNAMIC=1 flag causes make install to fail
-    CXX="g++ -fPIC" CC="gcc -fPIC" $MAKE DYNAMIC_ARCH=1 USE_OPENMP=1
+    if [[ ! $ARCH == 'aarch64' ]]; then
+        CXX="g++ -fPIC" CC="gcc -fPIC" $MAKE DYNAMIC_ARCH=1 USE_OPENMP=1
+    else
+        # openblas build fails with gcc-8.5, hence using gcc-8.4 explicitly
+        CXX="g++-8 -fPIC" CC="gcc-8 -fPIC" FC="gfortran-8" $MAKE DYNAMIC_ARCH=1 USE_OPENMP=1
```
Comment (leezu) [inline tools/dependencies/openblas.sh:40]: Have you tried updating openblas?

Comment (mseth10) [reply tools/dependencies/openblas.sh]: Tried OpenBLAS v0.3.10 (same version as master) and it works with gcc-8.5 , updated it. Thanks for the suggestion.

Comment (leezu) [reply tools/dependencies/openblas.sh]: Great! Would you consider v0.3.15? I see there are a couple of arm fixes in between of v0.3.10 and v0.3.15 including "Fixed segmentation faults in DYNAMIC_ARCH builds" which would affect us as we use DYNAMIC_ARCH=1

**Comment by Zha0q1** [comment]
Merging this pr for now so I can base the APL changes to this commit. We can update openblas in another pr

**Pull request merged**